### PR TITLE
Remove load ill indirect interface

### PR DESCRIPTION
**Pull request by gui-co**
**Description of work.**

This PR removes an old interface from Workbench. According to the code, this interface seems to be ILL only.

It is no longer used by the scientists here at ILL and the code contains several bugs.

**To test:**

Check that the interface is no longer in the menu:
Interfaces -> Indirect -> Tools

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by darominski — CHANGES_REQUESTED** [review]
Following the internal discussion at ILL, I can only approve the suppression of this unused and not functional code. 

However, it seems that there is more to be removed, namely `IndirectLoadILL.cpp` is still present at `qt/scientific_interfaces/Indirect/`, could you remove that as well or is there a reason to keep it? The documentation at `docs/source/interfaces/indirect/Indirect Tools.rst` is now obsolete. Also, could you add a release note?

**Comment by gui-co** [comment]
Thanks. I removed the remaining files and added a release note.

**Review by darominski — DISMISSED** [review]
Thanks, LGTM.

**Comment by martyngigg** [comment]
@Harrietbrown Can I just check ISIS don't use the transmission tab on the Indirect->Tools?

**Comment by gui-co** [comment]
Sorry, I did not see that the instrument list in the transmission tab was synchronized with the facility. If this tab is used, it will need a bit of refactoring; selecting an instrument different from IN16B (when ILL is the facility) causes a hard crash of Mantid.

**Comment by AnthonyLim23** [comment]
According to the current docs the transmission tab is used by ISIS and the SNS (https://docs.mantidproject.org/nightly/interfaces/indirect/Indirect%20Tools.html). So the best people to check with a probably @SpencerHowells and @peterfpeterson

**Comment by peterfpeterson** [comment]
@AndreiSavici is the better person to ask in this case

**Comment by yqchengornl** [comment]
The SNS BASIS team is using the indirect transmission tab. Is there a way to keep it? It can be moved and added to a different location in the menu if that makes the interface simpler. Thank you!

**Comment by SpencerHowells** [comment]
It looks as if it might not have read Efixed in the IPF. I've looked at the IPFs for IN10 & they seem Ok. On the ISIS Data Reduction interface we show the value of Efixed read & the box allows altering it. Perhaps Transmission should do something similar. It would also cure the problem of the instrument not having appropriate IPFs.

**Comment by gui-co** [comment]
Thanks all for the comments.

I can of course keep the transmission tab and only remove the load tab that is really ILL specific. Nevertheless, this transmission tab seems a bit redundant with the dialog of the IndirectTransmission algorithm. The only difference being the presentation of the output in the same window.

**IndirectTransmission dialog**
alg_dialog

**Indirect tool transmission tab**
interface

Let me know what you think. If the transmission tab is kept, i will fix the problem with non appropriate instruments.

**Comment by Harrietbrown** [comment]
We have several interfaces that function similarly, with them being a faceplate for an algorithm with an output visualiser. I think we will want to keep transmission.

**Comment by SpencerHowells** [comment]
I agree that we should keep Transmission interface

**Comment by yqchengornl** [comment]
I agree that keeping the transmission tab is a good idea. Showing the output in the same window instead of a workspace is more convenient for users.

**Comment by gui-co** [comment]
Transmission tab is restored.

I added a check on the instrument to avoid a hard crash. I also removed a non existing signal connection that was causing a confusing message in the terminal.

File: docs/source/interfaces/indirect/Indirect Tools.rst [hunk 1]
```diff
@@ -1,116 +0,0 @@
...
-Mass/Number Density
-  The density of the sample in :math: `g/cm^3` or :math:`atoms/\mathrm{\AA{}}^3`.
-
-Thickness
-  The thickness of the sample in :math:`cm`.
-
-Run
-  Runs the processing configured on the current tab.
-
-LoadILL
```
Comment (darominski) [inline docs/source/interfaces/indirect/Indirect Tools.rst:67]: Since the transmission tab is preserved only lines concerning LoadILL should be removed.

**Review by darominski — CHANGES_REQUESTED** [review]
I see now how the transmission could have been useful for other instruments, even though it seemed broken for the ILL, apologies. With the proposed changes, it looks good, except now for the too aggressive trimming of the Indirect Tools documentation. Could you please restore a part of it that refers to the transmission?

File: docs/source/interfaces/indirect/Indirect Tools.rst (hunk 1 above)
Comment (gui-co) [reply docs/source/interfaces/indirect/Indirect Tools.rst:67]: I restored the doc and removed this section.

**Review by darominski — APPROVED** [review]
Thank you for restoring the documentation, at this point this looks good to me.

**Comment by martyngigg** [comment]
Thanks for restoring the transmission tab 👍🏻

**Pull request merged**